app/src/main/java/ch/epfl/sweng/SDP/matchmaking/Matchmaker.java

```diff
@@ -1,73 +1,131 @@
 package ch.epfl.sweng.SDP.matchmaking;
 
-import android.support.annotation.NonNull;
-import android.util.Log;
-
-import ch.epfl.sweng.SDP.firebase.Database;
-import ch.epfl.sweng.SDP.firebase.Database.DatabaseReferenceBuilder;
-
 import com.google.firebase.database.DataSnapshot;
 import com.google.firebase.database.DatabaseError;
 import com.google.firebase.database.DatabaseReference;
 import com.google.firebase.database.ValueEventListener;
 
-import java.util.HashMap;
+import java.io.BufferedReader;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.net.URLEncoder;
 
-/**
- * Singleton enum responsible for the matchmaking.
- */
-public enum Matchmaker implements MatchmakingInterface {
-    // Singleton enum
+import ch.epfl.sweng.SDP.ConstantsWrapper;
 
-    INSTANCE;
+public class Matchmaker implements MatchmakingInterface {
 
-    private DatabaseReference myRef;
+    private ConstantsWrapper constantsWrapper;
+    private static Matchmaker singleInstance = null;
+    // static method to create instance of Singleton class
+    private DatabaseReference reference;
 
     /**
-     * Matchmaker init.
+     *  Create a singleton Instance.
+     * @return returns a singleton instance.
      */
-    Matchmaker() {
-        myRef = Database.INSTANCE.getReference("rooms");
+    public static Matchmaker getInstance(ConstantsWrapper constantsWrapper)
+    {
+        if (singleInstance == null) {
+            singleInstance = new Matchmaker(constantsWrapper);
+        }
+
+        return singleInstance;
+    }
+
+
+    /**
+     *  Matchmaker init.
+     */
+    private Matchmaker(ConstantsWrapper constantsWrapper) {
+
+        this.constantsWrapper = constantsWrapper;
+        this.reference = constantsWrapper.getReference("rooms");
 
         // Read from the database
-        myRef.addListenerForSingleValueEvent(new ValueEventListener() {
+        reference.addListenerForSingleValueEvent(new ValueEventListener() {
```

Comment:
What do you need this Listener for?

Reply:
Right now I only use this for debugging purpose. It is supposed to fetch all the rooms form the database. But since we don't need that right now you could delete it.

Reply:
Ok. I will leave it for now and would be very happy if you could do some manual testing to see if i broke anything.